src/models/regression.py

```diff
@@ -70,7 +70,8 @@ def train(self, num_epochs: int = 1,
                         batch_x[0].shape[0],
                         batch_x[0].shape[1] * batch_x[0].shape[2])
                     if self.include_pred_month:
-                        pred_months = batch_x[1]
+                        # @GABI does this do what you mean it to?
+                        pred_months = batch_x[1].astype(int)
```

Comment:
I am having troubles here because the `pred_months_onehot` won't merge with the 
in lines [78:80](https://github.com/esowc/ml_drought/pull/49/commits/d67e34acdddb00cdbde10252d81daf33a5dc7ff1#diff-0ae30feeeb42f997c0ae2d1a3d7a7254L77)
```python
                        x_in = np.concatenate(
                            (x_in, pred_months_onehot), axis=-1
                        )
```

because
```
ValueError: all the input arrays must have same number of dimensions
```

Reply:
```python
(Pdb) pred_months_onehot
array([], shape=(25, 0, 14), dtype=float64)

(Pdb) pred_months_onehot.shape
(25, 0, 14)

(Pdb) np.eye(14)[pred_months][:, 1:-1]
array([], shape=(25, 0, 14), dtype=float64)

(Pdb) pred_months
array([[12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1]])

(Pdb) pred_months.shape
(25, 2)

(Pdb) x_in.shape
(25, 70)

(Pdb) x_in = np.concatenate( (x_in, pred_months_onehot), axis=-1)
*** ValueError: all the input arrays must have same number of dimensions
```

Reply:
@gabrieltseng sorry a bit stuck on this one

Reply:
I think the issue is that the `pred_months` array now has 3 dimensions so when you select only the first and last indices (`np.eye(14)[pred_months][:, 1:-1]`) it removes all of those middle indices which relate to ?_each month_?

Reply:
Should pred_months be `(25, 2)` or should we use `np.unique(pred_months)` ?

Reply:
```python
(Pdb) np.eye(14)[np.unique(pred_months)].shape
(2, 14)

(Pdb) np.eye(14)[np.unique(pred_months)]
array([[0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0.]])

(Pdb) np.eye(14)[np.unique(pred_months)][:, 1:-1]
array([[1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.]])

(Pdb) np.eye(14)[np.unique(pred_months)][:, 1:-1].shape
(2, 12)
```

But that also doesn't work because:

```
(Pdb) np.concatenate((x_in, np.eye(14)[np.unique(pred_months)][:, 1:-1]), axis=-1)
*** ValueError: all the input array dimensions except for the concatenation axis must match exactly
(Pdb) x_in.shape
(25, 70)
(Pdb) np.eye(14)[np.unique(pred_months)][:, 1:-1]
array([[1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.]])
(Pdb) np.eye(14)[np.unique(pred_months)][:, 1:-1].shape
(2, 12)
```

Reply:
Why does `pred_months` have 3 dimensions?

Reply:
So when we have only one months worth of data:
```python
(Pdb) x_in.shape
(25, 36)

(Pdb) batch_x[1].astype(int)
array([12, 12, 12, 12, 12, 12, 12, 12, 12, 12, 12, 12, 12, 12, 12, 12, 12,
       12, 12, 12, 12, 12, 12, 12, 12])
(Pdb) batch_x[1].astype(int).shape
(25,)
```

the `pred_months` array is 1D

```python
(Pdb) np.eye(14)[pred_months][:, 1:-1]
array([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.]])
(Pdb) np.eye(14)[pred_months][:, 1:-1].shape
(25, 12)
```

But when we have two months of data
```python
(Pdb) batch_x[1].astype(int)
array([[12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1]])
(Pdb) pred_months.shape
(25, 2)
```

This code here creates a 3D `pred_months_onehot`

```python
(Pdb) pred_months_onehot = np.eye(14)[pred_months][:, :, 1:-1]
(Pdb) pred_months_onehot
array([[[0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
        [1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]],

       [[0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
        [1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]],

(Pdb) pred_months_onehot.shape
(25, 2, 12)
```

Reply:
Do we want `pred_months_onehot ` to look like:
```python
array([[1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
       [1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.],
...
```
If we have 2 months?

Reply:
I think the issue is that I don't know how to return the `pred_months` for multiple months:
Currently it is returned from the `chunk_array(train_dataloader[0], train_dataloader[1])[0]` 
as:
```python
(Pdb) batch_x[1].astype(int).shape
(25, 2)
(Pdb) batch_x[1].astype(int)
array([[12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1],
       [12,  1]])
```

If we want hte output to look like above (which i think is what we want) how do I add the arrays along the axis of interest?
```
# e.g.
(Pdb) a1=np.array([0,1,0,0])

(Pdb) a2=np.array([0,0,1,0])

(Pdb) a1
array([0, 1, 0, 0])

(Pdb) a2
array([0, 0, 1, 0])

(Pdb) a_ = np.vstack([a1,a2])

(Pdb) a_
array([[0, 1, 0, 0],
       [0, 0, 1, 0]])

(Pdb) a_.shape
(2, 4)
```

How do I add across that dimennsion 0 -> `(1,4)` or `(4,)`
```python
array([0, 1, 1, 0])
```